test/test_parse_credentials.ml

```diff
@@ -0,0 +1,29 @@
+open! Core
+open! Async
+open! Import
+open Connection.Credentials
+
+let%expect_test "parse password credentials" =
```

Comment:
I don't think we need to test the sexp (de)serialization. `ppx_sexp_conv` itself is extremely unlikely to introduce any breaking changes. If we want to guard against breaking changes to the format, we could use the stable module pattern seen [here](https://github.com/janestreet/async_smtp/blob/c2c1f8b7b27f571a99d2f21e8a31ce150fbd6ced/src/client_cache.ml).